src/systems/explicit_system.C

```diff
@@ -45,6 +46,8 @@ ExplicitSystem::~ExplicitSystem ()
 {
   // clear data
   this->clear();
+
+  remove_vector("RHS Vector");
```

Comment:
Why do we need this here?  The additional vectors are all about to be destroyed by System::~System anyways.